src/main/kotlin/app/revanced/patcher/apk/ApkBundle.kt

```diff
@@ -0,0 +1,186 @@
+import com.reandroid.arsc.util.FrameworkTable
+import com.reandroid.arsc.value.Entry
+import com.reandroid.common.TableEntryStore
+import com.reandroid.identifiers.TableIdentifier
+import java.io.File
+
+/**
+ * An [Apk] file of type [Apk.Split].
+ *
+ * @param files A list of apk files to load.
+ */
+class ApkBundle(files: List<File>) {
+    /**
+     * The [Apk.Base] of this [ApkBundle].
+     */
+    val base: Apk.Base
+
+    /**
+     * A map containing all the [Apk.Split]s in this bundle associated by their configuration.
+     */
+    val splits: Map<String, Apk.Split>?
+
+    init {
+        var baseApk: Apk.Base? = null
+
+        splits = buildMap {
+            files.forEach {
+                val apk = ApkModule.loadApkFile(it)
+                val (module, type) = ApkIdentifier.identifyModule(apk)
+                if (module is ApkIdentifier.Module.DynamicFeature) {
+                    return@forEach // Dynamic feature modules are not supported yet.
+                }
+
+                when (type) {
+                    ApkIdentifier.Type.Base -> {
+                        if (baseApk != null) {
+                            throw IllegalArgumentException("Cannot have more than one base apk")
+                        }
+                        baseApk = Apk.Base(apk)
+                    }
+
+                    is ApkIdentifier.Type.SplitConfig -> {
+                        val target = type.target
+                        if (this.contains(target)) {
+                            throw IllegalArgumentException("Duplicate split: $target")
+                        }
+
+                        val constructor = when (type) {
+                            is ApkIdentifier.Type.Asset -> Apk.Split::Asset
+                            is ApkIdentifier.Type.Library -> Apk.Split::Library
+                            is ApkIdentifier.Type.Language -> Apk.Split::Language
+                        }
+
+                        this[target] = constructor(target, apk)
+                    }
+                }
+            }
+        }.takeIf { it.isNotEmpty() }
+
+        base = baseApk ?: throw IllegalArgumentException("Base apk not found")
+    }
+
+    /**
+     * A [Sequence] yielding all [Apk]s in this [ApkBundle].
+     */
+    val all = sequence {
+        yield(base)
+        splits?.values?.let {
+            yieldAll(it)
+        }
+    }
+
+    /**
+     * Write all the [Apk]s inside the bundle to a folder.
+     *
+     * @param folder The folder to write the [Apk]s to.
+     * @return A sequence of the [Apk] files which are being refreshed.
+     */
+    internal fun write(folder: File) = all.map {
+        val file = folder.resolve(it.toString())
+        var exception: Apk.ApkException? = null
+        try {
+            it.write(file)
+        } catch (e: Apk.ApkException) {
+            exception = e
+        }
+
+        SplitApkResult(it, file, exception)
+    }
+
+    inner class ResourceTable internal constructor() {
```

Comment:
This class implements a lot of methods, that are not suitable in this class but instead in a seperate module that wraps around `arsclib`. Please move as much as `arsclib` related code to its own module.